server/data/restClient.ts

```diff
@@ -45,6 +47,38 @@ export default class RestClient {
     this.agent = config.url.startsWith('https') ? new HttpsAgent(config.agent) : new Agent(config.agent)
   }
 
+  async delete({
+    path = '',
+    query = '',
+    headers = {},
+    responseType = '',
+    raw = false,
+  }: DeleteRequest): Promise<unknown> {
+    logger.info(`Delete using user credentials: calling ${this.name}: ${path} ${query}`)
+    try {
+      const result = await superagent
+        .delete(`${this.apiUrl()}${path}`)
+        .agent(this.agent)
+        .use(restClientMetricsMiddleware)
+        .retry(2, (err, res) => {
+          if (err) logger.info(`Retry handler found API error with ${err.code} ${err.message}`)
+          return undefined // retry handler only for logging retries, not to influence retry logic
+        })
```

Comment:
bear in mind that this means it retries twice unconditionally so if you don't want that behaviour, you might want to mimic the `retry = false` thingy in `requestWithBody`.

Reply:
Yeah, I think for now we're happy to keep this for GET and DELETE. It was a useful addition to the template though, so we'll be sure to steal it (and your other refactors) if our needs change!

Reply:
i tend to cherry-pick updates to the template back to our repos, but it can be a total pain if we've changed a lot. i think ideally there could have been a base npm package with this kind of stuff.

>happy to keep this for GET and DELETE

at present, the (possibly-surprising!) retry behaviour is in DELTE but not in GET. it's in both in the template 🤷

Reply:
You mean in our repo? We have retries on GET here too

Yeah I think the optional approach is good! A straight cherry pick wouldn't quite here because we have hard mode on (`strict: true`), while the template doesn't 😅

Reply:
Yeah, we added it early on in the project. It was tricky getting everything to work, but not nearly as bad as adding it in much further down the line!